Islandora-Devops/isle-dc
# Add set-codebase-owner function

## This should add a function to set permissions of "codebase/" during the build process.
This also adds a function that you're able to call with a make command at any time. 

The command uses `find` with a couple of exclusions for improved performance of running a permissions change.
* Excludes the `/sites/default/files/` to prevent causing issues with File systems like S3.
* Excludes files with the proper permissions already.

During the build process, the vendor directory may already be there or is set to root, or in some cases, the entire codebase directory is root (this happens in some MACs due to user IDs).

This shouldn't impact any existing installations.

## How to test
1. Run `make clean` (to start with a completely clean build). 
2. Run `make set-codebase-owner`
 a. The codebase should not exist at this point so that it will output the following error message. "No codebase/ folder found, skipping"
3. Run `make starter-dev`
 b. This should build as normal. Check that the build is completed as normal.
